scipy/signal/windows/windows.py

```diff
@@ -1640,6 +1641,60 @@ def exponential(M, center=None, tau=1., sym=True):
     return _truncate(w, needs_trunc)
 
 
+def taylor(N, nbar=4, sll=-30):
+    """
+    Return the Taylor window.
+
+    The Taylor window allows for a selectable sidelobe suppression with a 
+    minimum broadening. This window is commonly used in synthetic aperture radar
+    processing.
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an
+        empty array is returned.
+    nbar : int
+        Number of nearly constant level sidelobes adjacent to the mainlobe
+    sll : float
+        Desired peak sidelobe level in decibels (db) relative to the mainlobe
+
+    Returns
+    -------
+    out : array
+        The window, with the maximum value normalized to one (the value
+        one appears only if the number of samples is odd).
+
+    See Also
+    --------
+    kaiser, bartlett, blackman, hamming, hanning
+
+    Notes
+    -----
+    The Kaiser window is described by Carrara, Goodman, and Majewski 
+    in 'Spotlight Synthetic Aperture Radar: Signal Processing Algorithms'
+    Page 512-513
+    """
+    B = 10**(-sll/20)
+    A = np.log(B + np.sqrt(B**2 - 1))/np.pi
+    s2 = nbar**2 / (A**2 + (nbar - 0.5)**2)
+    ma = np.arange(1,nbar)
+    def calc_Fm(m):
+        numer = (-1)**(m+1) * np.prod(1-m**2/s2/(A**2 + (ma - 0.5)**2))
+        denom = 2* np.prod([1-m**2/j**2 for j in ma if j != m])
+        return numer/denom
+    
+    Fm = np.array([calc_Fm(m) for m in ma])
+    def W(n):
+        return 2*np.sum(Fm * np.cos(2*np.pi*ma*(n-N/2 + 1/2)/N)) + 1
+    
+    w = np.array([W(n) for n in range(N)])
+    # normalize (Note that this is not described in the original text)
+    scale = 1.0/W((N-1)/2)
+    w *= scale
+    return w
```

Comment:
Can you ensure that for odd `N` exactly one value obtains a value of `1` and for even `N` none do? See e.g. #2221

Reply:
Yes, this is the logic.

Reply:
Testing other implementations of the Taylor Window they do not normalize/scale with a single value of 1. Does this requirement still stand?

Reply:
This ensures the requirements @larsoner . The explanation is that when we use `(N - 1)/2` as the argument for `n` in `W` we get
```
((N - 1)/2 - (N/2) + 1/2) == (N - 1 - N)/2 + 1/2 == -1/2 + 1/2 == 0
```
Which makes 0 the argument of the cosine function which will give us the maximum Taylor window value for a single point. This will be exactly equal to the largest value of the Taylor window when N is odd (as `n - N/2 + 1/2` will evaluate to 0 again when n is one less than the halfway point (which is why it was chosen)), and when N is even summing over the integers for `n` this will never become 0 (just +/- 1/2).